tools/src/main/scala/scala/scalanative/linker/LinktimeValueResolver.scala

```diff
@@ -17,6 +17,12 @@ trait LinktimeValueResolver { self: Reach =>
       s"$linktimeInfo.isLinux" -> Platform.isLinux,
       s"$linktimeInfo.isMac" -> Platform.isMac,
       s"$linktimeInfo.isFreeBSD" -> Platform.isFreeBSD,
+      s"$linktimeInfo.arch" -> {
+        config.compilerConfig.configuredOrDetectedTriple
+          .split('-')
```

Comment:
This might be problematic in the case of architectures that define subparts, e.g. arm, aarch64, but at this point when we mainly target x86 it should be enough. However we might need to ensure stability of `x86` which on some platforms is defined as `x86` on others as `i386` based on the source code from the LLVM I would opt for the latter

Reply:
I see you are way ahead of me.